Wikipedia/Code/DiffContainerViewController.swift

```diff
@@ -1227,34 +1236,41 @@ extension DiffContainerViewController: DiffToolbarViewDelegate {
     private func performUndo() {
         guard let wkProject = wkProject,
               let title = articleTitle,
-        let revisionID = toModelRevisionID,
-        let username = toModel?.user,
-        let summary = undoAlertSummaryTextField?.text else {
+              let revisionID = toModelRevisionID,
+              let username = toModel?.user,
+              let summary = undoAlertSummaryTextField?.text else {
             return
         }
-        
         fakeProgressController.start()
-        WKWatchlistService().undo(title: title, revisionID: UInt(revisionID), summary: summary, username: username, project: wkProject) { [weak self] result in
-
-            DispatchQueue.main.async {
-                self?.completeRollbackOrUndo(result: result, isRollback: false)
+        DispatchQueue.main.async {
```

Comment:
Threading error upon this undo as well - please move this `DispatchQueue.main.async` to only surround `self?.completeRollbackOrUndo` (line 1251).

Reply:
Updated all those!